onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -0,0 +1,287 @@
+                xform.user.username, xform.id_string, include_images=False
+            )
+            data = csv_df_builder._format_for_dataframe(
+                DataInstanceSerializer(instances, many=True).data
+            )
+            data = replace_special_characters_with_underscores(data)
+
+        return Response(data)
+
+    def partial_update(self, request, *args, **kwargs):
+        if request.user.is_anonymous():
+            return Response(
+                'Authentication credentials required.',
+                status.HTTP_400_BAD_REQUEST
+            )
+
+        self.object = self.get_object()
+        results = self.get_data(update=True)
+        if results.error:
+            return Response(results.message, status.HTTP_400_BAD_REQUEST)
+
+        if results.data:
+            serializer = OpenDataSerializer(self.object, data=results.data)
+
+            if serializer.is_valid():
+                _open_data = serializer.save()
+                if _open_data:
+                    return Response(
+                        "Record was successfully updated.",
+                        status.HTTP_200_OK
+                    )
+            else:
+                return Response(
+                    str(serializer.errors), status.HTTP_400_BAD_REQUEST
+                )
+
+        return Response(status=status.HTTP_200_OK)
+
+    def destroy(self, request, *args, **kwargs):
+        if request.user.is_anonymous():
+            return Response(
+                'Authentication credentials required.',
+                status.HTTP_400_BAD_REQUEST
+            )
+
+        self.get_object().delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @detail_route(methods=['GET'])
+    def schema(self, request, **kwargs):
+        self.object = self.get_object()
+        if isinstance(self.object.content_object, XForm):
+            xform = self.object.content_object
+            self.xform_headers = replace_special_characters_with_underscores(
+                xform.get_headers(), list_of_dicts=False
+            )
+
+            xform_json = json.loads(xform.json)
+            self.flatten_xform_columns(
+                json_of_columns_fields=xform_json.get('children')
+            )
+
+            tableau_column_headers = self.get_tableau_column_headers()
+
+            data = {
+                'column_headers': tableau_column_headers,
+                'connection_name': "%s_%s" % (
+                    xform.project_id, xform.id_string
+                ),
+                'table_alias': xform.title
+            }
+
+            return Response(data=data, status=status.HTTP_200_OK)
+
+        return Response(status=status.HTTP_404_NOT_FOUND)
+
+    @list_route(methods=['GET'])
+    def uuid(self, request, *args, **kwargs):
+        clazz = request.query_params.get('clazz')
+        object_id = request.query_params.get('object_id')
+
+        if not clazz or not object_id:
```

Comment:
The validation functionality from the above link is used when you are using a serializer class unless I misunderstood the docs. This action doesn't use the serializer class anywhere so I don't think we'll need the serializer's validation functionality or may be I'm missing the point.